sig-cli/contributing.md

```diff
@@ -0,0 +1,160 @@
+# Contributing to Sig-Cli
+
+This document explains the process for contributing code to the kubernetes
+repo through the sig-cli community.
+
+## TL;DR
+
+- New contributors should reach out on slack sig-cli - [link](https://github.com/kubernetes/community/tree/master/sig-cli) for an issue to work on that has an approved design
+- New contributors should attend the next sig-cli meeting and introduce themselves
+- Write tests
+
+## Message to new contributors
+
+Welcome to the Kubernetes sig-cli contributing guide.  We are excited
+about the prospect of you joining our community.  Please understand
+that all contributions to Kubernetes require time and commitment from
+the project maintainers to review the ux, software design, and code.  We
+recommend that you reach out to one of the sig-leads
+and ask the best way to get started.  It is likely that there a couple
+issues perfect for a new contributor to get started on.
+
+Sig-leads can be reached in the sig-cli slack channel or at the sig-cli
+bi-weekly meeting.  See the [sig-cli community page](https://github.com/kubernetes/community/tree/master/sig-cli)
+for a link to the slack channel, list of sig-leads, and the community meeting time and location.
+
+Example message for asking how to become a contributor:
+
+> Hello, my name is <your name>.  I would like to get involved in
+contributing to the Kubernetes project.  What is the best way to
+get started?
+
+## Life of a sig-cli feature
+
+Picking up an issue and implementing it without getting approval for
+the user experience and software design often results in wasted time
+and effort due.  Decisions such as flag-names, command names, specific
+command behavior.
+
+In order to minimize wasted work and improve communication across efforts,
+the user experience and software design must be agreed upon before
+any PRs are sent for code review.
+
+1. Identify a problem - GitHub issue with basic description
+2. Propose a solution - GitHub design proposal PR
+  - Mention `@kubernetes/sig-cli-misc`
+  - Expect feedback from 2-3 different sig-cli community members in the form of PR comments
+  - Incorporate feedback from sig-cli community members and comment "PTAL" (please take another look)
+  - Proposal accepted by at least one sig-cli lead
+  - Proposal merged
+3. Communicate what will be done - Discussion at sig-cli bi-weekly meeting - See the [sig-cli community page](https://github.com/kubernetes/community/tree/master/sig-cli) for meeting time and location.
```

Comment:
Should we suggest contributors to already go ahead and add their topic in the Agenda/Minutes doc for the date they are expecting to attend? Or maybe we just get used to always add a "contributor proposal" topic to every mtg, reserve some time for it, and do the call for everyone with related topics at the meeting?